Jenkins_jobs/WorkspaceInfo.groovy

```diff
@@ -158,10 +124,47 @@ def beautify(machineName, workspaceDirectory, workspaceStats, subdirectories) {
         }
     }
     output += "\n=======================================================================\n\n";
+    return output
 }
 
-
-timeout(time: params.timeout, unit: 'HOURS') {
-    parallel clones
+def errorMessage(String machineName, String message) {
+    String errorOutput = "\n\n****NEEDS_ATTENTION****"
+    errorOutput += "\n${machineName} failed"
+    errorOutput += "\nMessage:\n${message}"
+    errorOutput += "\n****NEEDS_ATTENTION****\n\n"
+    return errorOutput
 }
 
+def setupParallelPipelines(
+        timeOut,
+        machineName,
+        workspaceDirectory,
+        workspaceStatscmd,
+        subdirectoryStatscmd,
+        statOutputHeading
+        ) {
+
+    String currentMachine = env.NODE_NAME
+    clones[machineName] = {
+        try {
+            timeout(time: timeOut.getTime(), unit: 'HOURS') {
+                node (machineName) 
+                {                    
+                    String workspaceStats = sh (
+                            script: workspaceStatscmd,
+                            returnStdout: true).trim()
+
+                    String subdirectories = sh (
+                            script: subdirectoryStatscmd,
+                            returnStdout: true).trim()
+
+                    println beautify(statOutputHeading, workspaceDirectory, workspaceStats,subdirectories)
+                }
+            }
+        } catch(err) {
+            error (errorMessage(machineName, err.message))
+        } finally {
+            cleanWs()
```

Comment:
Seems we are only catching errors so we can cleanup and print them in a more pretty way? What are we cleaning up?

Reply:
Originally the idea was to print errors at the end, as a summary. But ran into issues with it so now it's printing the error with a keyword to make it easier to `Crtl+f` the log for errors. There's also the Blue Ocean plug-in if they don't want to sift through the log for errors.

The clean up is anything that may have been put on the disk during the execution. Not sure what but it's there just for good measures.

Reply:
My point is, if all we're doing is making the error message pretty, and the cleanWs isn't really doing anything, then we don't really need the try-catch-finally at all.